Feature/webclient/websockets

simaovii opened the pull request.

simaovii committed:
Add automatic saving to localstorage beta

simaovii committed:
Add SimplePoint object to represent a single point with own style
Remove not used dependencies
Comment ToolsConfig class
Corrrect few bugs

simaovii committed:
update package.json

simaovii committed:
add Persist class to handle websockets connections

simaovii committed:
Merge branch 'develop' into feature/webclient/upgradeview

simaovii committed:
Merge branch 'feature/webclient/upgradeview' into develop

simaovii committed:
merge with develop branch

simaovii committed:
add Support to Web Sockets
user can share board
it may have bugs

c-petterf committed:
fixed dependency injection

simaovii committed:
add first steps to persist data with web sockets and combine it with local storage

simaovii committed:
Add first steps to implement websockets

simaovii committed:
Merge branch 'feature/webapi/review-dependency-injection' into feature/webclient/websockets

simaovii committed:
Correct some bugs in websockets
Sync model with server
Correct bugs on server

PedDavid commented on WebClient/app/model/Grid.js:
```diff
@@ -1,8 +1,9 @@
 
 export default function Grid (initialFigures, currIdx) {
-  let figures = initialFigures    // TODO(simaovii): Change to hashmap
+  let figures = []    // TODO(simaovii): Change to hashmap
```
initialFigures not used?

simaovii replied:
são usadas lá em baixo, deixa estar como estava se não, não dá

PedDavid commented on WebClient/app/scene/Board/components/Modals/ShareBoardModal/index.jsx:
```diff
@@ -0,0 +1,34 @@
+// @flow
```
Should not be used!

PedDavid commented on WebClient/app/scene/Board/components/Modals/ShareBoardModal/index.jsx:
```diff
@@ -0,0 +1,34 @@
+import SignInModal from './components/SignInModal'
+import SignUpModal from './components/SignUpModal'
+
+export default class EnterUserModal extends React.Component {
```
Why not a functional component?

PedDavid commented on WebClient/app/util/Persist.js:
```diff
@@ -0,0 +1,45 @@
+// @Flow
```
Not needed!

PedDavid commented on WebClient/app/scene/Board/index.jsx:
```diff
@@ -7,56 +12,65 @@ import styles from './styles.scss'
+    grid.draw(this.canvasContext, 1)
+  }
+
+  componentWillMount () {
```
Não gosto nada!

PedDavid commented on WebClient/app/util/Persist.js:
```diff
@@ -0,0 +1,45 @@
+  connect = function (boardId) {
+    const scheme = document.location.protocol === 'https:' ? 'wss' : 'ws'
+    const port = document.location.port ? (53379) : ''
+    const connectionUrl = `${scheme}://${document.location.hostname}:${port}/api/websockets?id=${this.boardId}`
```
document.location.hostname is wrong

PedDavid commented on WebClient/app/util/Persist.js:
```diff
@@ -0,0 +1,45 @@
+  }
+}
+
+export function PersistType () {
```
??

PedDavid committed:
Remove flow anotations and update yarn.lock

PedDavid committed:
Change wrong endpoint, TODO for configuration

Usage of document.location.hostname was wrong (this is the hostname
location of the webclient server, not the api/websockets server).

All these values should be easy to reconfigure and together in some kind
of configuration file. Added TODO for this purpose.

simaovii committed:
Correct a bug with initial grid instance
A temporal Initial grid instance was being passed to default tools and when actual grid was created, temporal tool's grid was not updated
